packages/toolpad-core/src/types.ts

```diff
@@ -153,6 +154,8 @@ export type PropValueTypes<K extends string = string> = Partial<{
   [key in K]?: PropValueType;
 }>;
 
+export type ComponentProps = Record<string, unknown>;
+
 export interface ArgTypeDefinition<V = unknown> {
```

Comment:
Can we do
```suggestion
export interface ArgTypeDefinition<V = unknown, P = unknown> {
```
and pass `P` in
```tsx
export type ArgTypeDefinitions<P = any> = {
  [K in keyof P & string]?: ArgTypeDefinition<P[K], P>;
};
```
?

Reply:
Initialising `P` as `unknown` creates a bunch of `P is not assignable to unknown` errors. 

Initialising `P` as `any`
```ts 
export interface ArgTypeDefinition<V = unknown, P = any>
``` 
fixes things, since `P` is initialised as `any` here: 

```ts
export type ArgTypeDefinitions<P = any> = {
  [K in keyof P & string]?: ArgTypeDefinition<P[K], P>;
};
```

Reply:
Well yes, `any` can "fix" everything, that's why avoid using it 😄